babel/numbers.py

```diff
@@ -325,7 +325,8 @@ def get_decimal_symbol(locale: Locale | str | None = LC_NUMERIC) -> str:
 
     :param locale: the `Locale` object or locale identifier
```

Comment:
I think we will want a `numbering_system: str | None = None` parameter for this function (and the other similar functions). 

It should probably support both explicit numbering system names as well as the aliases (`'default'`, `'traditional'`), and if a specific numbering system is explicitly requested and it is not available in the locale, we should raise an exception instead of returning some default.

As you rightly did point out,

> Is it okay that this changes the numbering system used for formatting symbols from `latn` to the default numbering system for locales which default system isn't `latn`?

is a bit of a question. Looking at the data, 
```
>>> import collections
>>> collections.Counter(x.values())
Counter({'latn': 733, 'arab': 29, 'arabext': 19, 'deva': 13, 'adlm': 13, 'beng': 8, 'olck': 3, 'cakm': 3, 'tibt': 2, 'mymr': 2})
```
this would affect about 90 locales (21 macrolocales); I'd _love_ some input from `{'sd', 'dz', 'my', 'lrc', 'bn', 'ar', 'sa', 'raj', 'ccp', 'bgc', 'ks', 'bho', 'ps', 'sat', 'as', 'ne', 'mr', 'ckb', 'mzn', 'mni', 'fa'}` speakers for whether it would be weird to use non-latin symbols in conjuction with latin numerals. (On that note, could we have a test that proves this would now happen for e.g. `ar` locales?)

Reply:
> I'd love some input from {'sd', 'dz', 'my', 'lrc', 'bn', 'ar', 'sa', 'raj', 'ccp', 'bgc', 'ks', 'bho', 'ps', 'sat', 'as', 'ne', 'mr', 'ckb', 'mzn', 'mni', 'fa'} speakers for whether it would be weird to use non-latin symbols in conjuction with latin numerals.

Good question. Let's see if can get some native person to comment on that. I'll try to ask around in my circles.

I would like to later add a function like `format_digits(to_numbering_system, from_numbering_system="latn")` to convert digits between different locales. Do you think that it should be eventually integrated to other formatting numbers (like `format_decimal`)? If yes, then this question will become pointless as both numerals and symbols would be using same numbering system.

Reply:
> On that note, could we have a test that proves this would now happen for e.g. ar locales?

I'm not sure if I understand what kind of test you would like to have. But yup it's now mixing non-latin symbols and latin numerals.

Reply:
> It will lead to changing quite many functions when adding same parameter also to all functions that calls `get_decimal_symbol` (eg. `format_decimal` calls `NumberPattern.apply` which calls `get_decimal_symbol`) and other similar functions edited here. But it's probably the right thing to do.

Yeah, I think that's how it'll need to be. Oh, now that we're on this, I think the new parameter should definitely be kwarg-only! 

An additional thought: we could add a new API `get_numeric_symbols` which would return a dict (or preferably a dataclass) that'd contain all of the symbols for a given locale/system configuration, so we wouldn't need to call the small functions over and over again.

> It could be due to `jpan` being algorithmic system. Maybe not a problem but it's good to notice even with aliases it could raise an error. https://github.com/unicode-org/cldr/blob/main/common/supplemental/numberingSystems.xml#L45

Yeah, I think it's okay to let the KeyError bubble up if there are no digits or symbols given. We might want to document that in a docstring.

> On the other note language `hi` defines default and native numbering systems but there's symbols for over 40 numbering systems: https://github.com/unicode-org/cldr/blob/main/common/main/hi.xml#L5696-L6404

Most of those are `draft` status though, so we don't support them. Phew! ;)

> I would like to later add a function like `format_digits(to_numbering_system, from_numbering_system="latn")` to convert digits between different locales.

I have the feeling there might be subtleties there even with a non-algorithmic digit system. I haven't (yet...) read the CLDR spec about digit systems closely enough to say whether a simple translation can ever be enough; again, I think we would really benefit of a subject matter expert in e.g. `ar` or other systems 🤔 

> I'm not sure if I understand what kind of test you would like to have. But yup it's now mixing non-latin symbols and latin numerals.

Yeah, basically currently just a test that would show that you (could) get the non-Latin symbol with Latin numerals when this is merged.

Reply:
> An additional thought: we could add a new API get_numeric_symbols which would return a dict (or preferably a dataclass) that'd contain all of the symbols for a given locale/system configuration, so we wouldn't need to call the small functions over and over again.

Hmm to which level functions would the dataclass be passed? I'm not sure how nice the api would have to pass the dataclass to eg. `format_decimal`.